openmdao/recorders/tests/test_sqlite_reader.py

```diff
@@ -3055,6 +3056,89 @@ def test_constraints_with_aliases(self):
+        # recording was done with a module that is not available while doing the 
+        # reading of the case recorder file
+        
+        module_path = 'mymodule.py'  # the module file that will exist while recording but not reading
+        
+        def create_case_recording_file():
+            # Define the module content as a string
+            module_content = """
+class DummyClass(object):
+    pass
+            """
+            # Write the content to the file to create the module
+            with open(module_path, 'w') as file:
+                file.write(module_content)
+
+            # need to do this because the use_tempdirs decorator does not 
+            #  update the python path and so "." is not included and this 
+            #  module being created cannot be found
+            current_dir = os.getcwd()
+            if current_dir not in sys.path:
+                sys.path.append(current_dir)
+
+            # import the newly created module
+            import mymodule
+            
+            class ParaboloidWithDummyMetadata(om.ExplicitComponent):
+                def setup(self):
+                    self.add_input('x', val=0.0)
+                    self.add_input('y', val=0.0)
+                    self.add_output('f_xy', val=0.0)
+                def setup_partials(self):
+                    self.declare_partials('*', '*', method='fd')
+                def compute(self, inputs, outputs):
+                    x = inputs['x']
+                    y = inputs['y']
+                    outputs['f_xy'] = (x - 3.0)**2 + x * y + (y + 4.0)**2 - 3.0
+                def initialize(self):
+                    self.options.declare('dummy', types=mymodule.DummyClass, default=mymodule.DummyClass())
+           
+            prob = om.Problem()
+            recorder = om.SqliteRecorder('cases.sql')
+            prob.add_recorder(recorder)
+
+            prob.model.add_subsystem('parab_with_dummy_metadata', ParaboloidWithDummyMetadata(), 
+                                     promotes_inputs=['x', 'y'])
+            prob.model.add_subsystem('const', om.ExecComp('g = x + y'), promotes_inputs=['x', 'y'])
+            prob.model.set_input_defaults('x', 3.0)
+            prob.model.set_input_defaults('y', -4.0)
+            prob.driver = om.ScipyOptimizeDriver()
+            prob.driver.options['optimizer'] = 'COBYLA'
+            prob.model.add_design_var('x', lower=-50, upper=50)
+            prob.model.add_design_var('y', lower=-50, upper=50)
+            prob.model.add_objective('parab_with_dummy_metadata.f_xy')
+            prob.model.add_constraint('const.g', lower=0, upper=10.)
+            prob.setup()
+            prob.run_driver()
+            prob.record('final')
+            
+            prob.cleanup()
+            
+            # Really remove the module!
+            del sys.modules['mymodule']
+            os.remove(module_path)
+
+
+        create_case_recording_file()
```

Comment:
Instead of adding current dir to sys.path inside of `create_case_recording_file`, you might consider doing it here and have a try: finally block around the function call so you can restore sys.path back to its original form.

Reply:
Good idea. Changed